File: src/main/java/personal/ffivquizbot/slash_command/job_question/JobQuestionProvider.java
package personal.ffivquizbot.slash_command.job_question;

import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Random;

@Service
public class JobQuestionProvider {

    public ArrayList<FFIVJobs> getRandomJobList(Integer elementCount) {
        ArrayList<FFIVJobs> jobListCopy = new ArrayList<>(Arrays.asList(FFIVJobs.values()));

        Random random = new Random();

        ArrayList<FFIVJobs> extractedRandoms = new ArrayList<FFIVJobs>();

        for (int i = 0; i < elementCount; i++) {
            int randomIndex = random.nextInt(jobListCopy.size());

            extractedRandoms.add(jobListCopy.get(randomIndex));

            jobListCopy.remove(randomIndex);
        }

        return extractedRandoms;
    }
}

File: src/main/kotlin/personal/ffivquizbot/slashcommand/jobquestion/JobQuestionProvider.kt
package personal.ffivquizbot.slashcommand.jobquestion

import org.springframework.stereotype.Service
import personal.ffivquizbot.slashcommand.jobquestion.enums.FFIVJobs

@Service
class JobQuestionProvider {
    fun getRandomJobList(elementCount: Int): List<FFIVJobs> {
        return FFIVJobs.values().toList().shuffled().take(elementCount)
    }
}
